File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/realm/models/ResolvedVariable.java
package ch.rmy.android.http_shortcuts.realm.models;

import io.realm.RealmObject;

public class ResolvedVariable extends RealmObject {

    private String key;
    private String value;

    public static ResolvedVariable createNew(String key, String value) {
        ResolvedVariable resolvedVariable = new ResolvedVariable();
        resolvedVariable.setKey(key);
        resolvedVariable.setValue(value);
        return resolvedVariable;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/realm/models/ResolvedVariable.kt
package ch.rmy.android.http_shortcuts.realm.models

import io.realm.RealmObject
import io.realm.annotations.Required

open class ResolvedVariable : RealmObject() {

    @Required
    var key: String = ""
    @Required
    var value: String = ""

    companion object {

        fun createNew(key: String, value: String): ResolvedVariable {
            val resolvedVariable = ResolvedVariable()
            resolvedVariable.key = key
            resolvedVariable.value = value
            return resolvedVariable
        }
    }
}
